### PR TITLE
Update to .NET 6

### LINKED ISSUE #306: Target .net 6 on template
The dotnet new gtkapp template generates project file for .net 5. I think it is good to change it to .net 6 by default since it is the latest LTS now?

### PR DESCRIPTION
**Pull request by kant2002**
I update not only samples, but also codegen tools, because .NET 5 is not LTS and in 4 month would be out.
Added template parameters, and resurrect netcoreapp3.1 support in templates.
Tested with following process
```
dotnet new --install Source/Templates/GtkSharp.Template.CSharp
...
dotnet new gtkapp -t net5.0
```

Closes #306

### CODE REVIEW

**Review by harry-cpp — CHANGES_REQUESTED** [review]
Looks good, however the template .net target also needs to be bumped up, they are found in: https://github.com/GtkSharp/GtkSharp/tree/develop/Source/Templates

**Review by harry-cpp — APPROVED** [review]

File: Source/Templates/GtkSharp.Template.CSharp/content/GtkSharp.Application.CSharp/.template.config/template.json [hunk 1]
```diff
@@ -18,8 +18,27 @@
   "symbols": {
     "targetframework": {
       "type": "parameter",
-      "defaultValue": "net5.0",
-      "replaces": "net5.0"
+      "description": "The target framework for the project.",
+      "datatype": "choice",
```
Comment (harry-cpp) [inline Source/Templates/GtkSharp.Template.CSharp/content/GtkSharp.Application.CSharp/.template.config/template.json:22]: Hmm, will this lock us into selecting only one of the three possible frameworks, or can we still specify any target framework during creation and this just helps with the documentation for the command?

Comment (kant2002) [reply Source/Templates/GtkSharp.Template.CSharp/content/GtkSharp.Application.CSharp/.template.config/template.json:22]: I believe this is list of all possible values. Do you need additional frameworks? I list all except .Net Framework family here, can add that too.

Comment (harry-cpp) [reply Source/Templates/GtkSharp.Template.CSharp/content/GtkSharp.Application.CSharp/.template.config/template.json:22]: I would like to leave it as it was, as when .net 7 releases, you will be able to create the project with that as the target framework, without any update to the templates.

**Pull request merged**